xpartition.py

```diff
@@ -338,15 +380,32 @@ def __init__(self, xarray_obj):
     def initialize_store(self, store: str):
         self._obj.to_zarr(store, compute=False)
 
-    def write(self, store: str, ranks: int, dims: Sequence[Hashable], rank: int):
-        _write_partition_dataset(self._obj, store, ranks, dims, rank)
+    def write(
+        self,
+        store: str,
+        ranks: int,
+        dims: Sequence[Hashable],
+        rank: int,
+        collect_variable_writes: bool = False,
```

Comment:
Maybe add a docstring? It's hard to understand this option name without more context. Maybe it would be clearer if passed to the constructor of `Map` instead of changing the `write` API.

Reply:
Agreed, a docstring would be helpful here; I'll go ahead and add one.  Could you explain more regarding what you mean by passing it to the constructor of `Map`?  `Map` is currently not used when calling `ds.partition.write`.

Reply:
Sorry I meant to add at as an argument here: https://github.com/spencerkclark/xpartition/blob/56e9e31d545ac443efc13148a31394fadbcd522a/xpartition.py#L377

I'm not sure if that's possible with accessors (I must admit I don't see much value to accessors compared to plain functions/classes)

Reply:
I see.  Indeed, yeah, that is not possible with accessors.  The use of accessors was mostly just an aesthetic choice -- I found them to be a nice way to collect related DataArray or Dataset functionality under a particular namespace.  E.g. being able to call `da.partition.write` or `ds.partition.write` feels cleaner than importing separate functions, or writing a single overloaded function that can act on either DataArrays or Datasets.  

I'll probably go ahead and merge this for now.  We can always redesign things later if we'd like.

Reply:
No need to change now, but I give some more arguments against accessors in #15